restapi/stories.go

```diff
@@ -34,8 +34,54 @@ func init() {
 
 func (api api) ReturnAllStories(w http.ResponseWriter, r *http.Request) render.Renderer {
 	var stories []models.Story
+	var storiesByTags []models.Tag
 
-	err := api.database.Where("is_visible = true").Find(&stories).Error
+	getVisibility := r.URL.Query()["visibility"]
+	visibility := "true"
+	if getVisibility != nil {
+		visibility = getVisibility[0]
+	}
+
+	r.ParseForm()
+	sort := r.Form["sort"]
+	order := r.Form["order"]
+	tags := r.Form["tags"]
+
+	sortString := ""
+	for i:=0; i < len(sort); i++ {
+		//special case for author name because our db table doesnt have "name" in one column
+		if(sort[i] == "author_name"){
+			sortString += "author_first_name" + " " + order[i] + ", " + "author_last_name" + " " + order[i]
+		}else {
+			sortString += sort[i] + " " + order[i]
+		}
+
+		if i != len(sort)-1{
+			sortString += ", "
+		}
+	}
+
+	gormStories := api.database.Table("stories")
+	gormTags := api.database.Table("tags")
+
+	err := gormTags.Where("name IN ?", tags).Find(&storiesByTags).Error
+	if err != nil {
+		return rest.ErrInternal(api.logger, err)
+	}
+
+	if len(storiesByTags) != 0 {
+		var storyIDs []string
+		for i:=0; i<len(storiesByTags); i++{
+			storyIDs = append(storyIDs, fmt.Sprint(storiesByTags[i].StoryID))
```

Comment:
We don't need to append every time, we already know the final length of the storyIDs array a head of time. We can just array index.

Also `strconv.Itoa` is how we convert between int to string.